### DISCUSSION TITLE
Is there an api that can read and save the content of a file type note？

### DISCUSSION THREAD

**Discussion by SiriusXT** [TriliumNext/Trilium#3880] [Q&A]
Is there an api that can read and save the content of a file type note, test(await window.note.getNoteComplement()).content is invalid.
I'm trying to integrate drawio with a widget, and I need to read and save drawio-type files as strings.

**Comment by soulsands** [comment]
It may be necessary to confirm whether your window.note is FNOTE so that getNoteComplement can be called.

**Reply by SiriusXT** [reply]
> It may be necessary to confirm whether your window.note is FNOTE so that getNoteComplement can be called.

 note comes from async refreshWithNote(note), the file type note test (await window.note.getNoteComplement()).content
is undefined

**Comment by shutaozhenzhen** [comment]
I think you can get note content from BNote.getContent. And you can create note from BackendScriptApi.createNewNote.
Then you can replace note with new note

**Reply by SiriusXT** [reply]
> I think you can get note content from BNote.getContent. And you can create note from BackendScriptApi.createNewNote. Then you can replace note with new note

note comes from async refreshWithNote(note), the file type note test getContent is undefined

**Reply by shutaozhenzhen** [reply]
?
where is refreshWithNote? I cannot find this api in current api document

**Comment by SiriusXT** [comment]
Currently implementing in svg format

**Comment by zadam** [comment]
`await note.getNoteComplement()` should work, but only for "string typed" notes which is detected from MIMEs. SVG is treated as a string type, drawio proprietary files probably not.
